superset/sql_lab.py

```diff
@@ -245,7 +252,19 @@ def execute_sql_statement(
                 query.id,
                 str(query.to_dict()),
             )
-            data = db_engine_spec.fetch_data(cursor, query.limit)
+            data = db_engine_spec.fetch_data(cursor, increased_query)
+            if query.limit is None or len(data) <= query.limit:
+                query.limiting_factor = LimitingFactor.NOT_LIMITED
+            else:
+                data = data[:-1]
```

Comment:
add comment here: `returning 1 row less then the query.limit`

Reply:
good call, added